src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  animate,
+  animateChild,
+  group,
+  query,
+  state,
+  style,
+  transition,
+  trigger,
+} from '@angular/animations';
+import { Component, Input, OnInit } from '@angular/core';
+import { BehaviorSubject, take } from 'rxjs';
+import { colormapConfigToLegend, Legend, Plan } from 'src/app/types';
+
+import { MapService } from './../../services/map.service';
+
+@Component({
+  selector: 'app-create-scenarios',
+  templateUrl: './create-scenarios.component.html',
+  styleUrls: ['./create-scenarios.component.scss'],
+  animations: [
```

Comment:
Up to you, animations can be put in a separate file.

Reply:
I made the button and panel content transitions reusable and put them in a separate file (since those are just simple color and opacity transitions, respectively) but kept the expand/collapse panel animation in this file (since it's customized and animates its children as well).